crates/partition-store/src/snapshots/metadata.rs

```diff
@@ -76,6 +76,37 @@ pub struct PartitionSnapshotMetadata {
     pub files: Vec<LiveFile>,
 }
 
+impl PartitionSnapshotMetadata {
+    pub fn validate(
+        &self,
+        cluster_name: &str,
+        cluster_fingerprint: Option<ClusterFingerprint>,
+    ) -> anyhow::Result<()> {
+        if cluster_name != self.cluster_name {
+            anyhow::bail!(
+                "Snapshot does not match the cluster name of latest snapshot at destination!. Expected: {cluster_name}, got: {}",
+                self.cluster_name
+            );
+        }
+
+        // If the snapshot doesn't have a fingerprint on it, we'll ignore the
+        // check. Eventually all new snapshots will have a fingerprint and this
+        // check will be kept for very old snapshots only.
```

Comment:
Could make it explicit that we expect snapshots produced by clusters running at least v1.5.0 will always have a fingerprint?